### PR TITLE
chore: fix ignored dialyzer warnings (part 1) (4.3)

### PR DESCRIPTION
**Pull request by thalesmg**
Leftovers:
- `emqx_connection:info/2` and `emqx_ws_connection:info/2`
  - These seem to have clauses that are only accessed in tests.  Maybe we could move those clauses to another function exported only for tests, `debug_info/2`?
- `emqx_auth_pgsql_cli:connect/1` :: apparently wrong typespec in the dep?
- `emqx_lwm2m_coap_resource:coap_discover/2` :: apparently wrong typespec in the behavior definition in the dep?

### CODE REVIEW

File: src/emqx_channel.erl [hunk 1]
```diff
@@ -1113,19 +1117,16 @@ handle_timeout(_TRef, Msg, Channel) ->
 %% Ensure timers
 %%--------------------------------------------------------------------
 
-ensure_timer([Name], Channel) ->
-    ensure_timer(Name, Channel);
-ensure_timer([Name | Rest], Channel) ->
-    ensure_timer(Rest, ensure_timer(Name, Channel));
```
Comment (thalesmg) [inline src/emqx_channel.erl:1119]: These clauses with list input were apparently never called.

File: src/emqx_channel.erl [hunk 2]
```diff
@@ -1140,16 +1141,13 @@ reset_timer(Name, Time, Channel) ->
...
 interval(alive_timer, #channel{keepalive = KeepAlive}) ->
     emqx_keepalive:info(interval, KeepAlive);
 interval(retry_timer, #channel{session = Session}) ->
     timer:seconds(emqx_session:info(retry_interval, Session));
 interval(await_timer, #channel{session = Session}) ->
-    timer:seconds(emqx_session:info(await_rel_timeout, Session));
-interval(expire_timer, #channel{conninfo = ConnInfo}) ->
-    timer:seconds(maps:get(expiry_interval, ConnInfo));
-interval(will_timer, #channel{will_msg = WillMsg}) ->
-    timer:seconds(will_delay_interval(WillMsg)).
```
Comment (thalesmg) [inline src/emqx_channel.erl:1152]: These two clauses of `interval/2` were never invoked, apparently.  These two timers are always called in `ensure_timer/3`, but not by `ensure_timer/2`, which is the place that consults the timer map.

File: src/emqx_channel.erl [hunk 3]
```diff
@@ -1757,8 +1755,6 @@ shutdown(success, Reply, Channel) ->
 shutdown(Reason, Reply, Channel) ->
     {shutdown, Reason, Reply, Channel}.
 
-shutdown(success, Reply, Packet, Channel) ->
```
Comment (thalesmg) [inline src/emqx_channel.erl:1760]: This arity-4 clause is apparently never called with `success`.

File: src/emqx_frame.erl [hunk 4]
```diff
@@ -788,8 +786,6 @@ serialize_utf8_pair({Name, Value}) ->
 serialize_binary_data(Bin) ->
     [<<(byte_size(Bin)):16/big-unsigned-integer>>, Bin].
 
-serialize_utf8_string(undefined, false) ->
```
Comment (thalesmg) [inline src/emqx_frame.erl:791]: Never called with `false` here.

File: src/emqx_plugins.erl [hunk 5]
```diff
@@ -105,7 +101,7 @@ unload(PluginName) when is_atom(PluginName) ->
             ?LOG(error, "Plugin ~s is not started", [PluginName]),
             {error, not_started};
         {_, _} ->
-            unload_plugin(PluginName, true)
```
Comment (thalesmg) [inline src/emqx_plugins.erl:108]: Always called with `true` in the second argument position.

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
great job!

**Pull request merged**